services/src/main/java/org/fao/geonet/api/AllRequestsInterceptor.java

```diff
@@ -92,6 +93,7 @@ private void createSessionForAllButNotCrawlers(HttpServletRequest request) {
             }
         } else {
             HttpSession httpSession = request.getSession(false);
+            ServiceContext.clear2();
```

Comment:
If this logic is specific to this one location I would prefer if it was explicitly doing the work and document what it was doing:
```
// Avoid maintaining thread local resource use for crawlers
ServiceContext.clearThreadLocal();
ApplicationContextHolder.clear();
```